extras/kms/option.go

```diff
@@ -0,0 +1,131 @@
+package kms
+
+import (
+	"io"
+
+	"github.com/hashicorp/go-dbw"
+)
+
+// getOpts - iterate the inbound Options and return a struct
+func getOpts(opt ...Option) options {
```

Comment:
At some point in the past you indicated some regret at not using `(*options, error)` as the pattern the way I did in the base package and wrappers in go-kms-wrapping. Not saying you need to do it, but I didn't think about it until just now and figured I'd remind you in case you did want to make that change.

Reply:
ty for bringing this up... I think, this is fine and I do understand the pros/cons of both (after implementing both).